File: I/232-implement-queue-using-stacks/main.go
package main

import (
	"fmt"
)

type MyQueue struct {
	Q []int
}

/** Initialize your data structure here. */
func Constructor() MyQueue {
	return MyQueue{Q: []int{}}
}

/** Push element x to the back of queue. */
func (this *MyQueue) Push(x int) {
	this.Q = append(this.Q, x)
}

/** Removes the element from in front of queue and returns that element. */
func (this *MyQueue) Pop() int {
	r := this.Q[0]
	this.Q = this.Q[1:]
	return r
}

/** Get the front element. */
func (this *MyQueue) Peek() int {
	return this.Q[0]
}

/** Returns whether the queue is empty. */
func (this *MyQueue) Empty() bool {
	return len(this.Q) == 0
}

func main() {
	fmt.Println("a")
}

File: I/232-implement-queue-using-stacks/main.rs
struct MyQueue {
  ls: Vec<i32>,
  rs: Vec<i32>,
}

impl MyQueue {
  fn new() -> Self {
    MyQueue {
      ls: Vec::new(),
      rs: Vec::new(),
    }
  }

  fn push(&mut self, x: i32) {
    self.ls.push(x);
  }

  fn pop(&mut self) -> i32 {
    if self.rs.len() == 0 {
      while self.ls.len() > 0 {
        self.rs.push(self.ls.pop().unwrap());
      }
    }
    self.rs.pop().unwrap()
  }

  fn peek(&mut self) -> i32 {
    let v = self.pop();
    self.rs.push(v);
    v
  }

  fn empty(&self) -> bool {
    self.ls.len() == 0 && self.rs.len() == 0
  }
}
